core/src/config.rs

```diff
@@ -89,19 +145,43 @@ impl Config {
     }
 }
 
-fn admin_public_key_from(
+/// Generate a Holo admin keypair from holochain public key, email and password
+pub fn admin_keypair_from(
     holochain_public_key: PublicKey,
     email: &str,
     password: &str,
-) -> Result<PublicKey, Error> {
+) -> Result<Keypair, Error> {
     // This allows to use email addresses shorter than 8 bytes.
     let salt = Sha512::digest(email.as_bytes());
-    let mut hash = [0; SEED_SIZE];
+    keypair_from(
+        password.as_bytes(),
+        &salt,
+        &holochain_public_key.to_bytes(),
+        ARGON2_ADDITIONAL_DATA
+    )
+}
 
+/// Most general function for generating Holo keypairs. 
+/// Prefer using a more specialised version (e.g. admin_keypair) if possible
+/// passphrase: bytes of a user readable passphrase to be hashed when deriving the key pair
+/// salt: Addtional salt to be added as part of the Argon2 algo
+/// extra_secret: Optional, pass empty byte array if not desired. See argon2min docs
+/// argon_additional_data: Optional, pass empty byte array if not desired. See argon2min docs
+pub fn keypair_from(
```

Comment:
https://github.com/Holo-Host/holo-config/pull/11#discussion_r339583864

Reply:
Yes, I included this for Willem's sake.  However, just asking every repository maker to just Remember "right, so I'm trying to generate an Admin key; what was the particular argon2 parameter set we agreed to use, in this version of Holo?  Let me type that in ..." is also not the right approach, I think.